pkg/tsdb/cloudwatch/models/cloudwatch_query_test.go

```diff
@@ -1187,11 +1189,82 @@ func Test_ParseMetricDataQueries_account_Id(t *testing.T) {
 				{
 					JSON: []byte(`{"accountId":"some account id", "statistic":"Average"}`),
 				},
-			}, time.Now(), time.Now(), false, false)
+			}, time.Now(), time.Now(), logger, false, false)
 		assert.NoError(t, err)
 
 		require.Len(t, actual, 1)
 		require.NotNil(t, actual[0])
 		assert.Nil(t, actual[0].AccountId)
 	})
 }
+
+func Test_ParseMetricDataQueries_ApplyMacros(t *testing.T) {
+	t.Run("should expand $__period_auto macro when a metric search code query is used", func(t *testing.T) {
+		testCases := []struct {
+			startTime      time.Time
+			expectedPeriod string
+		}{
+			{
+				startTime:      time.Now().Add(-2 * time.Hour),
```

Comment:
I don't think it's going to matter that much in this case, but it may be better practice in general to get `time.Now()` once, so the difference between the starttime and endtime is more consistent.